### PR TITLE
Fix for issue #686 List all public repositories

### PR DESCRIPTION
**Pull request by rms81**
Added the functionality and tests. Not sure if it can be used as is though.
The request takes to long to return. About 10 minutes for about 65K repositories in my tests.
Maybe this could be done passing the pagination ability to the caller of the method.

### CODE REVIEW

File: Octokit.Tests.Integration/Clients/RepositoriesClientTests.cs [hunk 1]
```diff
@@ -534,6 +534,19 @@ public class TheGetMethod
         }
     }
 
+    public class TheGetAllPublicMethod
+    {
+        [IntegrationTest]
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/RepositoriesClientTests.cs]: If you could throw a `Skip` on this test, it'd be greatly appreciated - this test is going to run for a *long, long, long* time

**Comment by shiftkey** [comment]
>  Maybe this could be done passing the pagination ability to the caller of the method.

This is something I need to think about again. I should open a discussion somewhere on this...

**Comment by shiftkey** [comment]
This looks great! :metal: If you can mute that slow test I'm happy to merge this in!

File: Octokit.Tests.Integration/Clients/RepositoriesClientTests.cs (hunk 1 above)
Comment (rms81) [inline Octokit.Tests.Integration/Clients/RepositoriesClientTests.cs]: Of course. I will do that.

**Comment by rms81** [comment]
Done

**Pull request merged**